web/controllers/repository_controller.ex

```diff
@@ -45,10 +45,14 @@ defmodule Beanie.RepositoryController do
   end
 
   defp repository_list("true") do
-    %{"repositories" => from_docker} = Beanie.RegistryAPI.catalog(Beanie.registry)
-    Beanie.Repository.Query.update_list(from_docker)
-    # TODO refresh repository listing, then fetch from db
-    {:ok, Repo.all(Repository), :updated}
+    %{"repositories" => from_docker, :error => error} = Beanie.RegistryAPI.catalog(Beanie.registry)
```

Comment:
If you modify `Beanie.RegistryAPI.catalog/1` to return `{:ok, respositories}` or `{:error, error}`, then you can just pass the error straight on.